c2corg_ui/static/js/imageuploader.js

```diff
@@ -204,16 +203,22 @@ app.ImageUploaderController.prototype.upload_ = function() {
       });
 
       this.getImageMetadata_(file);
-      this.uploading[i] = this.api_.uploadImage(file);
-      this.promises.push(this.uploading[i]);
 
-      this.uploading[i].then(function(resp) {
+      var canceller = this.q_.defer();
+      var promise = this.api_.uploadImage(file, canceller.promise, function(file, event) {
+        var progress = event.loaded / event.total;
+        file['progress'] = 100 * progress;
+        console.log('progress', 100 * progress, file['metadata']['id']);
+      }.bind(this, file));
+      this.uploading[i] = promise;
+      this.cancellers.push(canceller);
+
+      promise.then(function(resp) {
         console.log('100% uploaded! ' + file['metadata']['title'] + ' ' + i);
       }.bind(this), function(resp) {
         this.alerts_.addError('error while uploading the image ' + resp);
-      }.bind(this), function(evt) { // handled by defer.notify()
-          //var progressPercentage = parseInt(100.0 * evt.loaded / evt.total, 10);
-      });
+        this.files.splice(this.files.indexOf(file), 1);
```

Comment:
Here I remove the file from files because we don't want to post an image on api that is not correctly uploaded on image backend.